migration/README.md

```diff
@@ -1,17 +1,84 @@
-# Migration steps
+# Upgrade and migration
 
-This directory contains the migration steps for each major and minor version, including the documentation with instructions and the snippets for automation of the upgrade process.
+This directory contains the upgrade and migration steps for each major and minor version, including the documentation with instructions and the snippets for automation of the upgrade process.
 
 > [!note]
 > Only the five latest major and minor releases are kept, if you want to see the migration guides for older versions then you need to check out that version of the repository.
 
-## Migration template and script
+## Writing migration steps for contributions
 
-> [!NOTE]
-> Only major releases are supported by the script
+Notice to developers on writing migration steps:
 
-There is a script that can help you create the initial migration folder and `upgrade-apps.md` document. You can use the following command to run it from the root of compliantkubernetes-apps directory:
+- Migration steps:
+    - are written per major and minor version and placed in a subdirectory of the migration directory with the name `main/`,
+        - during release this is promoted to the next release series `vX.Y/`,
+    - are written to be idempotent and usable no matter which patch version you are upgrading from and to,
+    - are documented in `main/README.md` to be able to run them manually,
+    - are divided into prepare and apply steps:
+        - Prepare steps:
+            - are placed in the `prepare/` directory,
+            - may **only** modify the configuration of the environment,
+            - may **not** modify the state of the environment,
```

Comment:
Not sure if we want to use `an environment` instead of `the environment`.
I'll let you decide.

```suggestion
            - may **only** modify the configuration of an environment,
            - may **not** modify the state of an environment,
```